### PR TITLE
Made the GMM Estimator take a seed and have configurable initialization

### PR DESCRIPTION
**Pull request by tomerk**

### CODE REVIEW

**Comment by etrain** [comment]
Ugh, left my comments on the commit rather than the "files changed" but this looks pretty good. If you could fix the few style things I pointed out that would be great. Thanks @tomerk !

**Comment by etrain** [comment]
Actually - while you're at it if you could add a test that runs on the data set I just posted to #200 and checks that the output of both implementations on that dataset are "close" to (0,0) for means and (25,1) or (1,25) for sigma, as well as close to 50/50 for the mixture weights, that would be great!

**Comment by etrain** [comment]
here "close" can be liberal.

**Comment by tomerk** [comment]
@etrain took care of all the above

File: src/main/scala/nodes/learning/GaussianMixtureModelEstimator.scala [hunk 1]
```diff
@@ -63,13 +63,42 @@ case class GaussianMixtureModelEstimator(
...
+
+      case RANDOM_INITIALIZATION =>
+        // Random Initialization
+        val colMin = min(X(::, *)).toDenseVector
+        val colMax = max(X(::, *)).toDenseVector
+        val colRange = colMax - colMin
+
+        val rand = new RandBasis(new ThreadLocalRandomGenerator(new MersenneTwister(seed)))
+
+        val gmmMeans = DenseMatrix.rand(k, numFeatures, Uniform(0, 1)(rand))
```
Comment (etrain) [inline src/main/scala/nodes/learning/GaussianMixtureModelEstimator.scala]: nice

File: src/test/scala/nodes/learning/GaussianMixtureModelSuite.scala [hunk 2]
```diff
@@ -90,6 +90,31 @@ class GaussianMixtureModelSuite extends FunSuite with LocalSparkContext with Log
...
+
+    val gmmEst = GaussianMixtureModelEstimator(k, minClusterSize = 1, seed = 0, stopTolerance = 0, maxIterations = 30)
+
+    val data = sc.parallelize(TestUtils.loadFile("gmm_data.txt"))
+    val parsedData = data.map(s => DenseVector[Double](s.trim.split(' ').map(_.toDouble))).cache()
+
+
+    val centers = DenseMatrix.zeros[Double](2, 2)
+
+    val variances = DenseMatrix((1.0, 25.0), (25.0, 1.0))
```
Comment (etrain) [inline src/test/scala/nodes/learning/GaussianMixtureModelSuite.scala]: Hmm.. is the fit going to be deterministically in this order?

**Comment by etrain** [comment]
All of this looks good to me except for the last comment about the test - or does the fixed seed just solve this?

**Comment by tomerk** [comment]
The deterministic seed solves the ordering.

**Pull request merged**